CommunityFridgeMapApi/tests/unit/test_tag.py

```diff
@@ -0,0 +1,62 @@
+import unittest
+from dependencies.python.db import Tag
+
+class DynamoDbMockPutItem():
+    def __init__(self):
+        pass
+
+    def put_item(self, TableName=None, Item=None, ConditionExpression=None):
+        pass
+
+class TagTest(unittest.TestCase):
+
+    def test_is_valid_tag_name(self):
+        tag = Tag(tag_name='test-tag', db_client=None)
+        self.assertTrue(tag.is_valid_tag_name('test-tag')[0])
+        self.assertEqual(tag.is_valid_tag_name('test-tag')[1], '')
+        tag = [
+            Tag(tag_name='', db_client=None),
+            Tag(tag_name=None, db_client=None),
+            Tag(tag_name='ab', db_client=None),
+            Tag(tag_name='test&tag', db_client=None),
+            Tag(tag_name='TEST-tag_tag123xyui1234567890__--', db_client=None)
+        ]
+        expected_false = [
+            tag[0].is_valid_tag_name('')[0],
+            tag[1].is_valid_tag_name(None)[0],
+            tag[2].is_valid_tag_name('ab')[0],
+            tag[3].is_valid_tag_name('test&tag')[0],
+            tag[4].is_valid_tag_name('TEST-tag_tag123xyui1234567890__--')[0]
+        ]
+        expected_equal = [
+            (tag[0].is_valid_tag_name('')[1], 'tag_name is an empty string'),
+            (tag[1].is_valid_tag_name(None)[1], 'tag_name is None'),
+            (tag[2].is_valid_tag_name('ab')[1], 'Length of the tag_name is 2. It should be >= 3 but <= 32.'),
+            (tag[3].is_valid_tag_name('test&tag')[1], 'tag_name contains invalid characters'),
+            (tag[4].is_valid_tag_name('TEST-tag_tag123xyui1234567890__--')[1],
+                'Length of the tag_name is 33. It should be >= 3 but <= 32.')
+        ]
+        for i in range(5):
+            self.assertFalse(expected_false[i])
+            self.assertEqual(expected_equal[i][0], expected_equal[i][1])
```

Comment:
nit: `self.assertEqual(expected_equal[i])`

Reply:
when I use this I get an Assertion Error stating that two arguments are required.

Reply:
I’m reviewing this on my tablet, so can’t see the entire codebase. Ideally `self.assertEqual` should also be able to de-structure a tuple as well. But this should be acceptable for now.